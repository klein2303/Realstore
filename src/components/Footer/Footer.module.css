.footer {
    width: 100%;
    height: fit-content;
    font-family: "Source Sans 3", sans-serif;
    margin-top: auto;
}

.imageholder {
    width: 100%;
    height: 350px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.informationbox {
    display: flex;
    flex-direction: column;
    justify-content: start;
}

.image {
    width: 100%;
    height: 350px;
    object-fit: cover;
    position: absolute;
    z-index: 10;
    filter: brightness(20%);
}

.box {
    z-index: 20;
    background-color: white;
    width: 450px;
    height: 60%;
    border-radius: 30px;
    gap: 40px;

    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}

.title {
    z-index: 20;
    font-size: 30px;
    color: white;
    margin-bottom: 10px;
}

.informationlink,
.informationlink:focus {
    text-decoration: none;
    color: #424242;
    font-size: 20px;
}

.informationlink:hover {
    text-decoration: underline;
}

.informationtext {
    margin-top: 8px;
    font-weight: 100;
    font-size: 16px;
}

.informationtitle {
    font-size: 20px;
    font-weight: 200;
    color: #424242;
}

.logoimg {
    width: 100px;
    height: auto;
}

.logoinformation {
    font-size: 12px;
    color: #424242;
}

.logobox {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.divider {
    height: 70%;
    width: 1px;
    background-color: #424242;
}

@media (max-width: 500px) {
    .imageholder {
        width: 100%;
        height: 200px;

        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }

    .divider {
        height: 80%;
        width: 1px;
        background-color: #424242;
    }

    .image {
        width: 100%;
        height: 200px;
        object-fit: cover;
        position: absolute;
        z-index: 10;
        filter: brightness(20%);
    }

    .box {
        z-index: 20;
        background-color: white;
        width: 300px;
        height: 62%;
        border-radius: 15px;
        gap: 20px;

        display: flex;
        flex-direction: row;
        justify-content: center;
        align-items: center;
    }

    .title {
        z-index: 20;
        font-size: 18px;
        color: white;
    }

    .informationlink,
    .informationlink:hover,
    .informationlink:focus {
        text-decoration: none;
        color: #424242;
        font-size: 12px;
    }

    .informationtext {
        margin-top: 6px;
    }

    .informationtitle {
        font-size: 14px;
        font-weight: normal;
        color: #424242;
    }

    .logoimg {
        width: 55px;
        height: auto;
    }

    .logoinformation {
        font-size: 10px;
        color: #424242;
    }
}
