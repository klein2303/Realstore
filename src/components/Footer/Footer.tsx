import {Link} from "react-router-dom";
import styles from "./Footer.module.css";
import footer from "/footer.svg";
import logo from "/Logo.svg";

const Footer = () => {
    return (
        <footer className={styles.footer} role="contentinfo" aria-label="Website Footer">
            <section className={styles.imageholder}>
                <img src={footer} alt="Decorative footer image" className={styles.image} />
                <p className={styles.title}>EXPLORE OUR CATALOG</p>
                <div className={styles.box} role="region" aria-label="Footer Navigation">
                    <div className={styles.informationbox}>
                        <p className={styles.informationtitle}>GENERAL</p>
                        <Link to={"/"} className={styles.informationlink}>
                            <p className={styles.informationtext} aria-label="Home Link">
                                Home
                            </p>
                        </Link>
                        <Link to={"/Shopping"} className={styles.informationlink}>
                            <p className={styles.informationtext}>Shop</p>
                        </Link>
                    </div>
                    <div className={styles.divider} role="separator" aria-hidden="true" />
                    <div className={styles.logobox} aria-label="Footer Logo and Information">
                        <Link to={"/"}>
                            <img
                                src={logo}
                                alt="RealStore logo"
                                className={styles.logoimg}
                                aria-label="RealStore Logo"
                            />
                        </Link>
                        <p className={styles.logoinformation}>@2024 RealStore.com</p>
                    </div>
                </div>
            </section>
        </footer>
    );
};

export default Footer;
