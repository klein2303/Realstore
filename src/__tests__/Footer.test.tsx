import {render, screen} from "@testing-library/react";
import {describe, it, expect} from "vitest";
import {MemoryRouter} from "react-router-dom";
import Footer from "../components/Footer/Footer";

describe("Footer component", () => {
    it("renders the footer with all elements", () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>,
        );

        // Verify the footer role and label
        const footer = screen.getByRole("contentinfo", {name: /website footer/i});
        expect(footer).toBeInTheDocument();

        // Verify the decorative image
        const decorativeImage = screen.getByAltText(/decorative footer image/i);
        expect(decorativeImage).toBeInTheDocument();

        // Verify the title text
        const title = screen.getByText(/explore our catalog/i);
        expect(title).toBeInTheDocument();

        // Verify the footer navigation region
        const navRegion = screen.getByRole("region", {name: /footer navigation/i});
        expect(navRegion).toBeInTheDocument();

        // Verify the general links
        const homeLink = screen.getByRole("link", {name: /home link/i});
        const shopLink = screen.getByRole("link", {name: /shop/i});
        expect(homeLink).toHaveAttribute("href", "/");
        expect(shopLink).toHaveAttribute("href", "/Shopping");

        // Verify the logo section
        const logoImage = screen.getByAltText(/realstore logo/i);
        expect(logoImage).toBeInTheDocument();
        const logoInfo = screen.getByText(/@2024 realstore.com/i);
        expect(logoInfo).toBeInTheDocument();
    });

    it("renders navigation links correctly", () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>,
        );

        // Check the "Home" link
        const homeLink = screen.getByRole("link", {name: /home link/i});
        expect(homeLink).toHaveAttribute("href", "/");

        // Check the "Shop" link
        const shopLink = screen.getByRole("link", {name: /shop/i});
        expect(shopLink).toHaveAttribute("href", "/Shopping");
    });

    it("renders the logo and company information", () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>,
        );

        // Verify the logo
        const logoImage = screen.getByAltText(/realstore logo/i);
        expect(logoImage).toBeInTheDocument();

        // Verify the company information text
        const companyInfo = screen.getByText(/@2024 realstore.com/i);
        expect(companyInfo).toBeInTheDocument();
    });

    it("matches snapshot", () => {
        const {container} = render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>,
        );
        expect(container).toMatchSnapshot();
    });
});
