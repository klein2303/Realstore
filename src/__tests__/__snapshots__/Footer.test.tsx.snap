// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > matches snapshot 1`] = `
<div>
  <footer
    aria-label="Website Footer"
    class="_footer_eb428f"
    role="contentinfo"
  >
    <section
      class="_imageholder_eb428f"
    >
      <img
        alt="Decorative footer image"
        class="_image_eb428f"
        src="/project1/footer.svg"
      />
      <p
        class="_title_eb428f"
      >
        EXPLORE OUR CATALOG
      </p>
      <div
        aria-label="Footer Navigation"
        class="_box_eb428f"
        role="region"
      >
        <div
          class="_informationbox_eb428f"
        >
          <p
            class="_informationtitle_eb428f"
          >
            GENERAL
          </p>
          <a
            class="_informationlink_eb428f"
            href="/"
          >
            <p
              aria-label="Home Link"
              class="_informationtext_eb428f"
            >
              Home
            </p>
          </a>
          <a
            class="_informationlink_eb428f"
            href="/Shopping"
          >
            <p
              class="_informationtext_eb428f"
            >
              Shop
            </p>
          </a>
        </div>
        <div
          aria-hidden="true"
          class="_divider_eb428f"
          role="separator"
        />
        <div
          aria-label="Footer Logo and Information"
          class="_logobox_eb428f"
        >
          <a
            href="/"
          >
            <img
              alt="RealStore logo"
              aria-label="RealStore Logo"
              class="_logoimg_eb428f"
              src="/project1/Logo.svg"
            />
          </a>
          <p
            class="_logoinformation_eb428f"
          >
            @2024 RealStore.com
          </p>
        </div>
      </div>
    </section>
  </footer>
</div>
`;
